app/app.go

```diff
@@ -561,35 +559,43 @@ func NewStrideApp(
 	)
 	icaModule := ica.NewAppModule(&app.ICAControllerKeeper, &app.ICAHostKeeper)
 
-	// Create the middleware stacks
+	// Define the IBC stacks
+	// Stack one (ICAHost Stack) contains:
+	// - IBC
+	// - ICAHost
+	// - base app
+	icaHostIBCModule := icahost.NewIBCModule(app.ICAHostKeeper)
 
-	// Stack one contains
+	// Stack two (ICAController Stack) contains
 	// - IBC
-	// - ICA
-	// - icacallbacks
+	// - ICAController (middleware)
+	// - icacallbacks (middleware)
 	// - stakeibc
 	// - base app
-	var icamiddlewareStack porttypes.IBCModule
-	icamiddlewareStack = icacallbacksmodule.NewIBCModule(app.IcacallbacksKeeper, stakeibcIBCModule)
-	icamiddlewareStack = icacontroller.NewIBCModule(app.ICAControllerKeeper, icamiddlewareStack)
-	icaHostIBCModule := icahost.NewIBCModule(app.ICAHostKeeper)
+	icaCallbacksStack := icacallbacksmodule.NewIBCModule(app.IcacallbacksKeeper, stakeibcIBCModule)
+	icaControllerStack := icacontroller.NewIBCModule(app.ICAControllerKeeper, icaCallbacksStack)
 
-	// Stack two contains
+	// Stack three (Records Stack) contains
 	// - IBC
-	// - records
+	// - records (middleware)
 	// - transfer
 	// - base app
 	recordsStack := recordsmodule.NewIBCModule(app.RecordsKeeper, transferIBCModule)
 
 	// Create static IBC router, add transfer route, then set and seal it
+	// Note, authentication module packets are routed to the top level of the middleware stack
 	ibcRouter := ibcporttypes.NewRouter()
 	ibcRouter.
```

Comment:
We should double check that changing the route order here won't cause unexpected changes in how packets are routed

Reply:
talked offline, ordering shouldn't matter since its just adding items to a map